### PR TITLE
Use pin_memory in forward_batch.init_new to reduce decoding latency

### PR DESCRIPTION
**Pull request by litmei**
## Motivation

In low-latency scenarios, there are substantial idle gaps between Decode phases.

## Modifications

Profiling analysis identified a Host-to-Device (H2D) synchronization bottleneck within forward_batch.init_new. Applying .pin_memory() converts this synchronous operation into an asynchronous one.

## Benchmarking and Profiling

Before:

image

After:

image

## Checklist

- [x] Format your code according to the [Format code with pre-commit](https://docs.sglang.io/developer_guide/contribution_guide.html#format-code-with-pre-commit).
- [ ] Add unit tests according to the [Run and add unit tests](https://docs.sglang.io/developer_guide/contribution_guide.html#run-and-add-unit-tests).
- [ ] Update documentation according to [Write documentations](https://docs.sglang.io/developer_guide/contribution_guide.html#write-documentations).
- [ ] Provide accuracy and speed benchmark results according to [Test the accuracy](https://docs.sglang.io/developer_guide/contribution_guide.html#test-the-accuracy) and [Benchmark the speed](https://docs.sglang.io/developer_guide/contribution_guide.html#benchmark-the-speed).
- [x] Follow the SGLang code style [guidance](https://docs.sglang.io/developer_guide/contribution_guide.html#code-style-guidance).

## Review Process

1. Ping Merge Oncalls to start the PR flow. See the [PR Merge Process](https://github.com/sgl-project/sglang/blob/main/.github/MAINTAINER.md#pull-request-merge-process).
2. Get approvals from [CODEOWNERS](https://github.com/sgl-project/sglang/blob/main/.github/CODEOWNERS) and other reviewers.
3. Trigger CI tests with [comments](https://docs.sglang.io/developer_guide/contribution_guide.html#how-to-trigger-ci-tests) or contact authorized users to do so.
   - `/tag-run-ci-label`, `/rerun-failed-ci`, `/tag-and-rerun-ci`
4. After green CI and required approvals, ask Merge Oncalls to merge.

### CODE REVIEW

File: python/sglang/srt/model_executor/forward_batch_info.py [hunk 1]
```diff
@@ -488,9 +488,9 @@ def init_new(
 
         num_tokens = len(batch.input_ids) if batch.input_ids is not None else 0
         if enable_num_token_non_padded(model_runner.server_args):
-            ret.num_token_non_padded = torch.tensor(num_tokens, dtype=torch.int32).to(
-                device, non_blocking=True
-            )
+            ret.num_token_non_padded = torch.tensor(
```
Comment (litmei) [inline python/sglang/srt/model_executor/forward_batch_info.py:492]: from sglang.srt.utils.common import is_pin_memory_available

_pin = is_pin_memory_available(device)

ret.num_token_non_padded = torch.tensor(
            num_tokens, dtype=torch.int32, pin_memory=_pin
        ).to(device, non_blocking=True)

Comment (litmei) [reply python/sglang/srt/model_executor/forward_batch_info.py:492]: fix Xeon platform

File: python/sglang/srt/model_executor/forward_batch_info.py [hunk 2]
```diff
@@ -806,10 +812,19 @@ def _compute_mrope_positions(
...
+                dim=1,
+            ).pin_memory()
+        else:
+            mrope_positions_cat = torch.cat(
+                [pos for pos in mrope_positions_list],
+                dim=1,
+            )
+        self.mrope_positions = mrope_positions_cat.to(
+            dtype=torch.int64, device=model_runner.device
+        )
```
Comment (litmei) [inline python/sglang/srt/model_executor/forward_batch_info.py]: could this part be simpler?

File: python/sglang/srt/model_executor/forward_batch_info.py [hunk 3]
```diff
@@ -63,7 +63,7 @@
     is_npu,
     support_triton,
 )
-from sglang.srt.utils.common import ceil_align
+from sglang.srt.utils.common import ceil_align, is_pin_memory_available
```
Comment (litmei) [inline python/sglang/srt/model_executor/forward_batch_info.py:66]: is_pin_memory_available -> add  torch.npu.is_available()
